projects/align-ptr.md

```diff
@@ -0,0 +1,39 @@
+# align(ptr)
+
+## Summary and problem statement
+
+The `align` attribute can be insufficiently expressive for some types, and we aim to improve some use cases.
+
+## Prioritization
+
+This group calls under the "Targeted ergonomic wins and extensions" priority.
+
+## Motivation, use-cases, and solution sketches
+
+It's sometimes the case that a type must be aligned based on the platform's pointer width. For example, the `AtomicPtr` type is defined as follows:
+
+```rust
+#[cfg_attr(target_pointer_width = "16", repr(C, align(2)))]
+#[cfg_attr(target_pointer_width = "32", repr(C, align(4)))]
+#[cfg_attr(target_pointer_width = "64", repr(C, align(8)))]
+pub struct AtomicPtr<T> {
+    p: UnsafeCell<*mut T>,
+}
+```
+
+That's three `cfg` lines without being able to say quite what we really want to say, "this is aligned to the size of a pointer".
+
+So far, the `align` attribute only allows for an integer literal as the input. This project group aims to add a new alignment input, `ptr`, which is always the value of the local pointer size.
+
+It has also been suggested that `size` be allowed as an input value, to request that a type have alignment equal to its own size.
```

Comment:
The `align` modifier only sets a *minimum* alignment. Oddly, Rust doesn't have an ergonomic way to specify the alignment of a type is *exactly* equal to a value.

You can set a maximum alignment with the `packed` modifier, but it's an error to have both `align` and `packed` annotations on the same type, even if they're not contradictory:
```
error[E0587]: type has conflicting packed and align representation hints
 --> src/main.rs:1:30
  |
1 | #[repr(align(1), packed(2))] pub struct Foo;
  |                              ^^^^^^^^^^^^^^^
```